src/main/java/analyzer/exercises/hamming/HammingAnalyzer.java

```diff
@@ -22,37 +22,11 @@ public void analyze(Solution solution, Analysis analysis) {
 
         solution.getCompilationUnits().forEach(cu -> cu.walk(ClassOrInterfaceDeclaration.class, walker));
 
-        if (!walker.hasConstructor()) {
-            analysis.addComment(new MustUseConstructor());
-            return;
-        }
-
-        if (!walker.constructorHasIfStatements() && !walker.constructorHasMethodCalls()) {
-            analysis.addComment(new MustUseConditionalLogicInConstructor());
-            return;
-        }
-
-        if (!walker.constructorThrowsIllegalArgument()) {
-            analysis.addComment(new MustThrowInConstructor());
-            return;
-        }
-
-        if (!walker.getHammingDistanceMethodMayCalculateDistance()
-                && !walker.constructorMayCalculateDistance()) {
-            analysis.addComment(new MustCalculateHammingDistance());
-            return;
-        }
-
         if (walker.usesCharacterLiterals()) {
             analysis.addComment(new AvoidCharacterLiterals());
             return;
         }
 
-        if (!walker.usesStringCharAtOrCodePointAt()) {
-            analysis.addComment(new MustUseStringCharAtOrCodePointAt());
-            return;
-        }
```

Comment:
Note that this comment is not redundant per se: It is possible to write a solution that passes the test and still receive this comment.

I don't think we should be very strict on the fact that an analyzer should _only_ contain feedback that is written down in the exercise's `.meta/design.md`. Rather I'd say that it should _at least_ contain that feedback, and it may contain more if the author thinks it's valid.

Reply:
Perfect, I like that!